### DISCUSSION TITLE
VTK returns a warning when assigning labels as point/cell data

### DISCUSSION THREAD

**Discussion by liberostelios** [pyvista/pyvista#1448] [Q&A] [answered]
When assigning labels as point/cell data, VTK prints the following warning message:

```
vtkDataSetAttributes.cx:1287  WARN| vtkPointData (0x7fcf2ce51170): Can not set attribute Scalars. Only vtkDataArray subclasses can be set as active attributes.
```

-----

**To Reproduce**
Just run the labels example from pyvista docs: https://docs.pyvista.org/examples/02-plot/labels.html

```py
# Make some random points
poly = pv.PolyData(np.random.rand(10, 3))

poly["My Labels"] = [f"Label {i}" for i in range(poly.n_points)]
```

**Screenshots**

The console shows this warning message:

<img width="1029" alt="Screenshot 2021-06-26 at 10 08 48" src="https://user-images.githubusercontent.com/1896327/123506842-83bad580-d666-11eb-8cb2-0963fc474afe.png">

-----

**System Information:**

```txt
--------------------------------------------------------------------------------
  Date: Sat Jun 26 10:10:10 2021 CEST

                OS : Darwin
            CPU(s) : 8
           Machine : x86_64
      Architecture : 64bit
       Environment : Python
        GPU Vendor : ATI Technologies Inc.
      GPU Renderer : AMD Radeon Pro 460 OpenGL Engine
       GPU Version : 4.1 ATI-4.5.14

  Python 3.9.5 (default, May  4 2021, 03:36:27)  [Clang 12.0.0
  (clang-1200.0.32.29)]

           pyvista : 0.31.1
               vtk : 9.0.20210612
             numpy : 1.20.3
           imageio : 2.9.0
           appdirs : 1.4.4
            scooby : 0.5.7
            meshio : 4.4.6
        matplotlib : 3.4.2
           IPython : 7.24.1
        ipyvtklink : 0.2.1
             scipy : 1.6.3
              tqdm : 4.61.1
--------------------------------------------------------------------------------
```

**Comment by tkoyama010** [comment, marked as answer]
I'm not sure about it, but the warning message seems to occur at VTK side.

https://github.com/Kitware/VTK/blob/861f0faa3aa8548d6d79fa565c27c366fba53713/Common/DataModel/vtkDataSetAttributes.cxx#L1285-L1287

Anyway,  `PolyData` item seems to set correctly. I think we don't have to care about it.

```python
>>> poly["My Labels"]
array(['Label 0', 'Label 1', 'Label 2', 'Label 3', 'Label 4', 'Label 5',
       'Label 6', 'Label 7', 'Label 8', 'Label 9'], dtype='<U7')
```

**Reply by akaszynski** [reply]
I've created an issue #1450 and PR #1451 to address this behavior.  As @tkoyama010 stated, the warning can be ignored, but I agree with @liberostelios, that the behavior is confusing.  Will be fixed in the next release.

**Reply by liberostelios** [reply]
Thank you so much for addressing this! Although the warning doesn't affect the functionality of `PolyData`, it breaks any kind of CLI when this occurs. For instance, I am using tqdm to show a progress bar in a script with a loop that constantly produces this warning and, quite obviously, the progress bar is broken due the messages spamming the terminal.

I am wondering about the fix, though. Will this affect plotting labels, as the data won't be set active anymore?
